rescript/orient.py

```diff
@@ -1,55 +1,68 @@
+                reference_sequences: DNAFASTAFormat = None,
                 perc_identity: float = 0.9,
                 query_cov: float = 0.9,
                 threads: int = 1,
                 left_justify: bool = False,
                 ) -> (DNAFASTAFormat, DNAFASTAFormat):
     matched_temp, notmatched = DNAFASTAFormat(), DNAFASTAFormat()
-    # use vsearch to search query seqs against reference database
-    # report orientation of query seqs relative to reference seqs.
-    with tempfile.NamedTemporaryFile() as out:
-        # note: qmask is disabled as DNAFASTAFormat requires all output seqs
-        # to be uppercase. Could loop through output seqs to convert to upper
-        # but which is faster: disabling masking or looping through with skbio?
-        cmd = ['vsearch', '--usearch_global', str(sequences),
-               '--matched', str(matched_temp), '--notmatched', str(notmatched),
-               '--db', str(reference_sequences), '--id', str(perc_identity),
-               '--maxaccepts', '1', '--strand', 'both', '--qmask', 'none',
-               '--query_cov', str(query_cov), '--threads', str(threads),
-               '--userfields', 'qstrand', '--userout', out.name]
-        if left_justify:
-            cmd.append('--leftjust')
-        run_command(cmd)
-        with open(out.name, 'r') as orient:
-            orientations = [line.strip() for line in orient]
+    if (reference_sequences is not None):
+        # use vsearch to search query seqs against reference database
+        # report orientation of query seqs relative to reference seqs.
+        with tempfile.NamedTemporaryFile() as out:
+            # note: qmask is disabled as DNAFASTAFormat requires all output
+            # seqs to be uppercase. Could loop through output seqs to convert
+            # to upper but which is faster: disabling masking or looping
+            # through with skbio?
+            cmd = ['vsearch',
+                   '--usearch_global', str(sequences),
+                   '--matched', str(matched_temp),
+                   '--notmatched', str(notmatched),
+                   '--db', str(reference_sequences),
+                   '--id', str(perc_identity),
+                   '--maxaccepts', '1',
+                   '--strand', 'both',
+                   '--qmask', 'none',
+                   '--query_cov', str(query_cov),
+                   '--threads', str(threads),
+                   '--userfields', 'qstrand',
+                   '--userout', out.name]
+            if left_justify:
+                cmd.append('--leftjust')
+            run_command(cmd)
+            with open(out.name, 'r') as orient:
+                orientations = [line.strip() for line in orient]
 
-    # if any query seqs are in reverse orientation, reverse complement
-    if '-' in orientations:
+        # if any query seqs are in reverse orientation, reverse complement
+        if '-' in orientations:
+            matched = DNAFASTAFormat()
```

Comment:
move these two lines to the top of the function outside of the conditional statement? then you avoid duplicating the lines below.

Reply:
That doesn't work. We only need to apply this if we're using reverse complementing against a reference. If thats the case, we check for a reverse orientation (line 51), then loop through the sequences (54-59) and if a given sequence is negative, reverse complement that. The if statements are on _slightly_ different variables. I could re-name the variables if that would help?

I think this is an indented version of your original logic, just brought under the "has reference" umbrella

Reply:
you're right. Technically ```matched = DNAFASTAFormat()``` could be moved outside of the logic, and it is repeated on line 63... but the way you have it is more readable anyway (better than saving 1 line of code 😆 )